File: android/app/src/main/java/com/workstationoffice/workcrm/kpi/DashboardRepository.kt
package com.workstationoffice.workcrm.kpi

import com.workstationoffice.workcrm.models.DashboardOverview
import com.workstationoffice.workcrm.networking.ApiClient
import java.time.Calendar
import java.time.YearMonth
import java.util.Date

object DashboardRepository {
    suspend fun overview(month: String? = null, repId: String? = null, teamId: String? = null): DashboardOverview =
        ApiClient.api.dashboardOverview(month = month, repId = repId, teamId = teamId)

    /** Mirrors iOS daysLeftInMonth — the kpi-alert cron's last-5-days rule. */
    fun daysLeftInMonth(now: Date = Date()): Int {
        val cal = java.util.Calendar.getInstance().apply { time = now }
        val lastDay = cal.getActualMaximum(java.util.Calendar.DAY_OF_MONTH)
        val today = cal.get(java.util.Calendar.DAY_OF_MONTH)
        return maxOf(0, lastDay - today + 1)
    }

    fun currentMonthKey(): String {
        val ym = YearMonth.now()
        return "%04d-%02d".format(ym.year, ym.monthValue)
    }
}

File: ios/WorkCRM/KPI/DashboardRepository.swift
import Foundation

public actor DashboardRepository {
    public static let shared = DashboardRepository()
    private init() {}

    public func overview(month: String? = nil, repId: String? = nil, teamId: String? = nil) async throws -> DashboardOverview {
        var query: [URLQueryItem] = []
        if let month  { query.append(URLQueryItem(name: "month",  value: month)) }
        if let repId  { query.append(URLQueryItem(name: "repId",  value: repId)) }
        if let teamId { query.append(URLQueryItem(name: "teamId", value: teamId)) }
        return try await APIClient.shared.get("dashboard/overview", query: query)
    }

    /// "5 working days left" header — same logic the backend's KPI-alert cron
    /// uses (`isLastFiveDaysOfMonth`). Computed client-side so the personal
    /// dashboard doesn't need to round-trip just to render a banner.
    public static func daysLeftInMonth(now: Date = Date()) -> Int {
        let cal = Calendar(identifier: .gregorian)
        guard let range = cal.range(of: .day, in: .month, for: now) else { return 0 }
        let lastDay = range.upperBound - 1
        let today = cal.component(.day, from: now)
        return max(0, lastDay - today + 1)
    }

    public static func currentMonthKey() -> String {
        let f = DateFormatter()
        f.locale = Locale(identifier: "en_US_POSIX")
        f.dateFormat = "yyyy-MM"
        return f.string(from: Date())
    }
}
